### PR TITLE
ACI Contract subject: Refactored

### PR DESCRIPTION
**Pull request by schunduri**

### CODE REVIEW

**Comment by dagwieers** [comment]
@schunduri It seems this branch contains various commits from another branch (most likely). As it includes aci_aep stuff as well. You can fix this using `git rebase -i HEAD~15` and removing those commits.

The likely cause is that you created a new branch, not from the master branch, but from you aci_aep feature branch.

**Comment by schunduri** [comment]
@dagwieers  Took care of it :)

**Comment by dagwieers** [comment]
Great, now you still need to fix those PEP8 problems ;-)
Also for the other PRs you did before.

File: library/aci_contract_subject.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_contract_subject
+short_description: Manages initial contract subjects(does not include contracts)
```
Comment (dagwieers) [inline library/aci_contract_subject.py]: Please make this: `Manage initial contract subjects on Cisco ACI fabrics`

File: library/aci_contract_subject.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+description:
+-  Manage a group of filters for a specific application or service using contract subjects in Cisco ACI APIC APIs with this module.
```
Comment (dagwieers) [inline library/aci_contract_subject.py]: Please make this: `Manage initial contract subjects on Cisco ACI fabrics.`

**Comment by schunduri** [comment]
@jmcgill298 Looks fine to me

@dagwieers on it

**Comment by jmcgill298** [comment]
@schunduri The filter_name and directive args should be removed from docs, and the module should provide a fail_msg when they are passed in pointing to the new aci_subject_bind_filter (willing to change to a better name).

**Comment by schunduri** [comment]
@jmcgill298 Thats great. I think we can call it aci_subject_filter_association / aci_subject_filter_binding and provide a detailed description

**Comment by schunduri** [comment]
@dagwieers the travis test fails cause of import issues in aci_rest? I dont understand why that is cause this file does not have any of those

**Comment by dagwieers** [comment]
@schunduri I think this is related to the history of the PR where i.e. wrong commits and forced pushes. I'll merge this as-is.

**Pull request merged**